internal/controllers/core/podlogstream/podlogstreamcontroller.go

```diff
@@ -158,16 +161,17 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	} else {
 		podNN := types.NamespacedName{Name: stream.Spec.Pod, Namespace: stream.Spec.Namespace}
 		pod, err := c.kClient.PodFromInformerCache(ctx, podNN)
-		if (err != nil && apierrors.IsNotFound(err)) ||
-			(pod != nil && pod.DeletionTimestamp != nil && !pod.DeletionTimestamp.IsZero()) {
+		deleting := false
+		if err != nil && apierrors.IsNotFound(err) {
 			c.deleteStreams(streamName)
+			deleting = true
 		}
 
-		if err != nil {
-			c.setErrorStatus(streamName, fmt.Errorf("streaming logs: %v", err))
-		} else if pod == nil {
-			c.setErrorStatus(streamName, fmt.Errorf("streaming logs: pod not found: %s", podNN))
-		} else {
+		if pod == nil || apierrors.IsNotFound(err) {
```

Comment:
I assume `pod == nil` whenever _any_ error is returned, no?
```suggestion
		if apierrors.IsNotFound(err) || (err == nil && pod == nil) {
```
(TBH the latter half seems unnecessary, as I'd assume it's an invariant of `PodFromInformerCache` that the Pod isn't `nil` if there's no error, so maybe `apierrors.IsNotFound(err)` is sufficient)

Reply:
yeah, you're right, but i think i can clean this up one better and merge it with the branch above. done!